### ISSUE TITLE
Does Code upload completely?

### ISSUE THREAD

**Issue by hoahoa1808** [MingkunLishigure/CACL#1]
@MingkunLishigure Sorry if my quest seem quite silly. I feel interesting with idea in the paper "Cluster-guided Asymmetric Contrastive Learning for Unsupervised Person Re-Identification" and want to reimplement your repos. however, I found these code has some segments are uncompleted.  for instance, the is no code for Processing class Or in Resnet class, `dn_x` is not assigned before used when i set `features 2048`. whether my confuse is true?

Second, in the trainer.py
```python
            flag = 0
            loss1 = self.memory1(bn_x1, full_conect2.clone(), indexes1, back = flag)
            flag = 1
            loss2 = self.memory2(bn_x2, full_conect1.clone(), indexes1, back = flag)
```
and in hm.py
```python
        if back == 0:
            return self.focal_loss( targets.detach().clone(),weight.detach().clone(),label_count.detach().clone(),sim.clone(),mask.clone())
        else:
            return self.focal_loss( targets.detach().clone(),weight.detach().clone(),label_count.detach().clone(),sim.clone(),mask.clone()) \
                +  self.contrasmemotyloss(targets.detach().clone(),sim.clone(),mask.clone(),old_inputs.detach().clone(),another_inputs_full.clone())  \
                +  self.contrasloss(targets.detach().clone(),sim.clone(),mask.clone(),old_inputs.detach().clone(),another_inputs_full.clone())
```
why loss1 has back 0 instead 1. i think loss1 is 0 and loss 2 is 1 because in paper said that "the parameters in the second network branch are updated only with the intra-views cluster-level contrastive loss"?

Finally, where is Predictor in paper theory in your code?

Thanks!

**Comment by MingkunLishigure** [comment]
Thanks for your quests!  I'm so sorry that we upload uncompleted code 2 months ago, we update the code about the data process and others.  For other questions, I list the response below:

1. “dn_x is not assigned before used when i set features 2048”： 
 The num_features default is 0, you do not need set 'num_features', once num_features > 0 it will cause this error.

2. In loss2, excluding the 'focal loss' part, the feature from the second network branch are all set as '.detach()', this loss gradient only feedbacks to the first network branch.

Tips：we have renewed the code in '/cacl'. This version can run with 'python SIC.py' without any additional parameter setting on dataset market-1501.
